### PR TITLE
bug 2117374: add message about working with restricted, but not restricted-v2

### PR DESCRIPTION
**Pull request by deads2k**
add message about working with restricted, but not restricted-v2

/assign @stlaz

### CODE REVIEW

**Comment by deads2k** [comment]
/hold cancel

added unit test.

**Comment by stlaz** [comment]
Looks good. I believe we only wanted this to be present in 4.11, please edit the PR to merge to the 4.11 branch instead of master.

File: pkg/securitycontextconstraints/sccadmission/admission.go [hunk 1]
```diff
@@ -315,6 +324,25 @@ loop:
 		}
 	}
 
+	// if we have restricted-v2, and we're not allowed (this means restricted-v2 did not match) and the user cannot use restricted-v1
+	// then we should check to see if restricted-v1 would allow the pod.  If so, prepend a specific failure message.
+	userCannotUseForRestricted := sets.NewString(denied...).Has("restricted")
```
Comment (ibihim) [inline pkg/securitycontextconstraints/sccadmission/admission.go:329]: mini nit: has/is are good indicators for a boolean value.

File: pkg/securitycontextconstraints/sccadmission/admission_test.go [hunk 2]
```diff
@@ -1135,6 +1135,96 @@ func TestAdmitPreferNonmutatingWhenPossible(t *testing.T) {
...
+	}
+
+	for testCaseName, testCase := range tests {
+		t.Run(testCaseName, func(t *testing.T) {
+			tc := setupClientSet()
+			lister := createSCCLister(t, testCase.sccs)
+			testAuthorizer := &sccTestAuthorizer{t: t}
+			plugin := newTestAdmission(lister, tc, testAuthorizer)
+
+			attrs := admission.NewAttributesRecord(testCase.newPod, testCase.oldPod, coreapi.Kind("Pod").WithVersion("version"), testCase.newPod.Namespace, testCase.newPod.Name, coreapi.Resource("pods").WithVersion("version"), "", testCase.operation, nil, false, &user.DefaultInfo{})
```
Comment (ibihim) [inline pkg/securitycontextconstraints/sccadmission/admission_test.go:1204]: nit: veeeeeeeeeeery long line 😄

File: pkg/securitycontextconstraints/sccadmission/admission.go [hunk 3]
```diff
@@ -315,6 +324,25 @@ loop:
...
+	hasRestrictedV2 := restrictedV2SCCProvider != nil
+	isAllowed := allowingProvider != nil
+	if hasRestrictedV2 && !isAllowed && userCannotUseForRestricted {
+		_, restrictedErrs := appliesToPod(restrictedSCCProvider, pod)
```
Comment (ibihim) [inline pkg/securitycontextconstraints/sccadmission/admission.go:333]: Could it happen that restrictedSCCProvider is not set?

Comment (deads2k) [reply pkg/securitycontextconstraints/sccadmission/admission.go:333]: No.  In order to be denied "restricted", the restricted SCC was iterated over in this loop `for i, provider = range providers {`

Comment (deads2k) [reply pkg/securitycontextconstraints/sccadmission/admission.go:334]: comment added.

File: pkg/securitycontextconstraints/sccadmission/admission.go (hunk 1 above)
Comment (deads2k) [reply pkg/securitycontextconstraints/sccadmission/admission.go:329]: > mini nit: has/is are good indicators for a boolean value.
`userCannotUseForRestricted` flows well in the `if` sentence below:
> if restrictedV2 exists and if the pod is not allowed and if the user cannot use restricted, then add this special message to the list of errors.

Comment (ibihim) [reply pkg/securitycontextconstraints/sccadmission/admission.go:329]: Well that reads surprisingly well. I didn't see it in the context of the if-statement.

**Pull request merged**